File: app/src/main/java/liuliu/hotel/model/CustomerModel.java
package liuliu.hotel.model;

import android.graphics.Bitmap;

import java.io.ByteArrayOutputStream;
import java.io.Serializable;

import liuliu.hotel.base.Base64Util;
import liuliu.hotel.utils.Utils;
import liuliu.hotel.web.globalfunc;

public class CustomerModel implements Serializable {
    private int id;
    //旅客信息
    private String SerialId;//流水号
    private String Name;//姓名
    private String Sex;//性别
    private String Nation;//民族
    private String Birthday;//出生日期
    private String CardType;//证件类型
    private String CardId;//证件号码
    private String Native;//籍贯
    private String Address;//地址
    private String RoomId;//房间号
    private String Area;//所属辖区
    private String CheckInTime;//入住时间
    private String CheckOutTime;//离店时间
    private String CheckInSign;//入住表示，当前程序版本号
    private Bitmap Headphoto;//头像
    private String Comment;//备注
    private String url;

    public CustomerModel() {

    }

    public CustomerModel(String name, String sex, String nation, String address, String roomId) {
        Name = name;
        Sex = sex;
        Nation = nation;
        Address = address;
        RoomId = roomId;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getUrl() {
        return url;
    }

    public void setUrl(String url) {
        this.url = url;
    }

    public String getXml(String xml, boolean isSave, DBLGInfo info) {

        //String xml = gfunc.getAssetsFileData("checkInNativeParameter.xml");

        if (xml != null && !xml.equals("")) {
            if (isSave) {
                xml = xml.replace("Method", "CheckInNative");
            } else {
                xml = xml.replace("Method", "CheckOutNative");
            }

            xml = xml.replace("InputHotelId", info.getLGDM());
            xml = xml.replace("InputAuthorizationCode",
                    info.getQYSCM());

            xml = xml.replace("InputSerialId", Utils.URLEncode(getSerialId()));
            xml = xml.replace("InputName", Utils.URLEncode(getName()));
            xml = xml.replace("InputSex", Utils.URLEncode(getSex()));
            xml = xml.replace("InputNation", Utils.URLEncode(getNation()));
            xml = xml.replace("InputBirthday", Utils.URLEncode(getBirthday()));
            xml = xml.replace("InputCardType", Utils.URLEncode(getCardType()));
            xml = xml.replace("InputCardId", Utils.URLEncode(getCardId()));

            xml = xml.replace("InputNative", Utils.URLEncode(getNative()));
            xml = xml.replace("InputAddress", Utils.URLEncode(getAddress()));
            xml = xml.replace("InputRoomId", Utils.URLEncode(getRoomId()));
            xml = xml.replace("InputCheckInTime", Utils.URLEncode(getCheckInTime()));
            xml = xml.replace("InputCheckOutTime", Utils.URLEncode(getCheckOutTime()));
            xml = xml.replace("InputReceiveTime", "");
            xml = xml.replace("InputCheckInSigen", Utils.URLEncode(getCheckInSign()));
            if (Headphoto != null) {
                ByteArrayOutputStream photo = new ByteArrayOutputStream();
                Headphoto.compress(Bitmap.CompressFormat.JPEG, 95, photo);
                byte[] ba = photo.toByteArray();
                String photodata = Base64Util.encode(ba);
                xml = xml.replace("InputHeadPhoto", photodata);
            } else {
                xml = xml.replace("InputHeadPhoto", "");
            }
            xml = xml.replace("InputComment", "");
        }

        return xml;
    }

    public String getLeaveXml(String xml) {
        if (!xml.equals("") && xml != null) {
            xml = xml.replace("InputSerialId", getSerialId());
            xml = xml.replace("InputLeaveTime", getCheckOutTime());
        }
        return xml;
    }

    public String getSerialId() {
        return SerialId;
    }

    public void setSerialId(String serialId) {
        SerialId = serialId;
    }

    public String getName() {
        return Name;
    }

    public void setName(String name) {
        Name = name;
    }

    public String getSex() {
        return Sex;
    }

    public void setSex(String sex) {
        Sex = sex;
    }

    public String getNation() {
        return Nation;
    }

    public void setNation(String nation) {
        Nation = nation;
    }

    public String getBirthday() {
        return Birthday;
    }

    public void setBirthday(String birthday) {
        Birthday = birthday;
    }

    public String getCardType() {
        return CardType;
    }

    public void setCardType(String cardType) {
        CardType = cardType;
    }

    public String getCardId() {
        return CardId;
    }

    public void setCardId(String cardId) {
        CardId = cardId;
    }

    public String getNative() {
        return Native;
    }

    public void setNative(String aNative) {
        Native = aNative;
    }

    public String getAddress() {
        return Address;
    }

    public void setAddress(String address) {
        Address = address;
    }

    public String getRoomId() {
        return RoomId;
    }

    public void setRoomId(String roomId) {
        RoomId = roomId;
    }

    public String getArea() {
        return Area;
    }

    public void setArea(String area) {
        Area = area;
    }

    public String getCheckInTime() {
        return CheckInTime;
    }

    public void setCheckInTime(String checkInTime) {
        CheckInTime = checkInTime;
    }

    public String getCheckOutTime() {
        return CheckOutTime;
    }

    public void setCheckOutTime(String checkOutTime) {
        CheckOutTime = checkOutTime;
    }

    public String getCheckInSign() {
        return CheckInSign;
    }

    public void setCheckInSign(String checkInSign) {
        CheckInSign = checkInSign;
    }

    public Bitmap getHeadphoto() {
        return Headphoto;
    }

    public void setHeadphoto(Bitmap headphoto) {
        Headphoto = headphoto;
    }

    public String getComment() {
        return Comment;
    }

    public void setComment(String comment) {
        Comment = comment;
    }
}

File: app/src/main/java/liuliu/hotel/model/CustomerModel.kt
package liuliu.hotel.model

import android.graphics.Bitmap

import java.io.ByteArrayOutputStream
import java.io.Serializable

import liuliu.hotel.base.Base64Util
import liuliu.hotel.method.*
import liuliu.hotel.method.DBFHInfo
import liuliu.hotel.utils.Utils
import liuliu.hotel.web.globalfunc

class CustomerModel : Serializable {
    var id: Int = 0
    //旅客信息
    var serialId: String? = null//流水号
    var name: String? = null//姓名
    var sex: String? = null//性别
    var nation: String? = null//民族
    var birthday: String? = null//出生日期
    var cardType: String? = null//证件类型
    var cardId: String? = null//证件号码
    var native: String? = null//籍贯
    var address: String? = null//地址
    var roomId: String? = null//房间号
    var area: String? = null//所属辖区
    var checkInTime: String? = null//入住时间
    var checkOutTime: String? = null//离店时间
    var checkInSign: String? = null//入住表示，当前程序版本号
    var headphoto: Bitmap? = null//头像
    var comment: String? = null//备注
    var url: String? = null

    constructor() {
        val fh = arrayOfNulls<DBFHInfo>(3)
    }

    constructor(name: String, sex: String, nation: String, address: String, roomId: String) {
        this.name = name
        this.sex = sex
        this.nation = nation
        this.address = address
        this.roomId = roomId
    }

    fun getXml(xml: String?, isSave: Boolean, info: DBLGInfo): String {
        var xml = xml

        //String xml = gfunc.getAssetsFileData("checkInNativeParameter.xml");

        if (xml != null && xml != "") {
            if (isSave) {
                xml = xml.replace("Method", "CheckInNative")
            } else {
                xml = xml.replace("Method", "CheckOutNative")
            }

            xml = xml.replace("InputHotelId", info.lgdm)
            xml = xml.replace("InputAuthorizationCode", info.qyscm)
            xml = xml.replace("InputSerialId", Utils.URLEncode(serialId))
            xml = xml.replace("InputName", Utils.URLEncode(name))
            xml = xml.replace("InputSex", Utils.URLEncode(sex))
            xml = xml.replace("InputNation", Utils.URLEncode(nation))
            xml = xml.replace("InputBirthday", Utils.URLEncode(birthday))
            xml = xml.replace("InputCardType", Utils.URLEncode(cardType))
            xml = xml.replace("InputCardId", Utils.URLEncode(cardId))
            xml = xml.replace("InputNative", Utils.URLEncode(native))
            xml = xml.replace("InputAddress", Utils.URLEncode(address))
            xml = xml.replace("InputRoomId", Utils.URLEncode(roomId))
            xml = xml.replace("InputCheckInTime", Utils.URLEncode(checkInTime))
            xml = xml.replace("InputCheckOutTime", Utils.URLEncode(checkOutTime))
            xml = xml.replace("InputReceiveTime", "")
            xml = xml.replace("InputCheckInSigen", Utils.URLEncode(checkInSign))
            if (headphoto != null) {
                val photo = ByteArrayOutputStream()
                headphoto!!.compress(Bitmap.CompressFormat.JPEG, 90, photo)
                val ba = photo.toByteArray()
                val photodata = Base64Util.encode(ba)
                xml = xml.replace("InputHeadPhoto", photodata)
            } else {
                xml = xml.replace("InputHeadPhoto", "")
            }
            xml = xml.replace("InputComment", "")
        }

        return xml
    }

    fun getLeaveXml(xml: String?): String {
        var xml = xml
        if (xml != "" && xml != null) {
            xml = xml.replace("InputSerialId", serialId)
            xml = xml.replace("InputLeaveTime", checkOutTime)
        }
        return xml
    }
}
